File: DecompiledGameScripts/cyberpunk/devices/various/serverNodeHealthChangeListener.swift

public class ServerNodeHealthChangeListener extends CustomValueStatPoolsListener {

  private let m_serverNode: wref<ServerNode>;

  public final static func Create(serverNode: ref<ServerNode>) -> ref<ServerNodeHealthChangeListener> {
    let instance: ref<ServerNodeHealthChangeListener> = new ServerNodeHealthChangeListener();
    instance.m_serverNode = serverNode;
    return instance;
  }

  public func OnStatPoolValueChanged(oldValue: Float, newValue: Float, percToPoints: Float) -> Void {
    this.m_serverNode.OnHealthChanged(newValue);
  }
}

File: Definitions/Game_Definitions/classes/ServerNodeHealthChangeListener.lua
---@meta
---@diagnostic disable

---@class ServerNodeHealthChangeListener : gameCustomValueStatPoolsListener
---@field serverNode ServerNode
ServerNodeHealthChangeListener = {}

---@return ServerNodeHealthChangeListener
function ServerNodeHealthChangeListener.new() return end

---@param props table
---@return ServerNodeHealthChangeListener
function ServerNodeHealthChangeListener.new(props) return end

---@param serverNode ServerNode
---@return ServerNodeHealthChangeListener
function ServerNodeHealthChangeListener.Create(serverNode) return end

---@param oldValue Float
---@param newValue Float
---@param percToPoints Float
function ServerNodeHealthChangeListener:OnStatPoolValueChanged(oldValue, newValue, percToPoints) return end

